modules/imgproc/src/deriv.cpp

```diff
@@ -920,6 +968,10 @@ void cv::Laplacian( InputArray _src, OutputArray _dst, int ddepth, int ksize,
         Mat kernel(3, 3, CV_32F, K[ksize == 3]);
         if( scale != 1 )
             kernel *= scale;
+
+        CV_OCL_RUN(_dst.isUMat() && _src.dims() <= 2,
+                   ocl_Laplacian3_8UC1(_src, _dst, ddepth, kernel, delta, borderType));
```

Comment:
@alalek , thanks for the good comments. I have updated the patch to make sure laplacian call opencl path  before IPP.